spanner/spansql/parser.go

```diff
@@ -1286,6 +1289,50 @@ func (p *parser) parseAlterTable() (*AlterTable, *parseError) {
 	}
 }
 
+func (p *parser) parseAlterDatabase() (*AlterDatabase, *parseError) {
+	debugf("parseAlterDatabase: %v", p)
+
+	/*
+			ALTER DATABASE database_id
+		    	action
+
+			action:
+		    SET OPTIONS ( optimizer_version = { 1 ...  2 | null },
+		                  version_retention_period = { 'duration' | null },
+		                  enable_key_visualizer = { true | false | null } )
+	*/
+
+	if err := p.expect("ALTER"); err != nil {
+		return nil, err
+	}
+	pos := p.Pos()
+	if err := p.expect("DATABASE"); err != nil {
+		return nil, err
+	}
+	// The following method is also valid for database names.
+	dbname, err := p.parseTableOrIndexOrColumnName()
```

Comment:
It's a bit different, but I think we should bother with that. The database ID cannot be ended with underscore. Also, it supports hyphens. https://cloud.google.com/spanner/docs/data-definition-language#database-id-names

Reply:
I've added a TODO for it. Currently, spansql does not apply these restrictions for any identifiers, and it would make more sense to pick that up for all identifiers in a separate PR.